crates/node/src/storage/rocksdb.rs

```diff
@@ -111,10 +115,16 @@ const ADDRESS_GEN_KEY_SEGMENT: &str = "address_gen";
 
 const OLD_DIFF_PREFIX: &str = "old";
 const NEW_DIFF_PREFIX: &str = "new";
+const MAX_CHUNK_SIZE: usize = 10_000_000;
 
 /// RocksDB handle
 #[derive(Debug)]
-pub struct RocksDB(rocksdb::DB);
+pub struct RocksDB {
+    /// Handle to the db
+    inner: rocksdb::DB,
+    /// Indicates if read only
+    read_only: bool,
```

Comment:
oh? we already have something like this. this should be atomic, though, no? and we should be making use of it, so we don't block further abci calls from making progress (as long as they don't involve writes)

Reply:
No, we have a read only access type to a read/write rocksdb handle. We need to have two rocksdb handles open concurrently and the rocksdb impl enforces that at most one is read/write by keeping an os lock.

Reply:
I'm not sure what you mean by atomic. And the reason for having two concurrent handles is precisely so we don't block other abci calls.